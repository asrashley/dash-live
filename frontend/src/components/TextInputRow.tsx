import { FormRow, FormRowProps } from "./FormRow";
import { TextInput, TextInputProps } from "./TextInput";

export type TextInputRowProps = Pick<FormRowProps, "name" | "label" | "text" | "error"> & TextInputProps;

export function TextInputRow({
  name,
  label,
  text,
  error,
  ...props
}: TextInputRowProps) {
  return (
    <FormRow name={name} label={label} text={text} error={error}>
      <TextInput name={name} error={error} {...props} />
    </FormRow>
  );
}
