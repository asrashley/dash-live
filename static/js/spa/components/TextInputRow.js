import { html } from "htm/preact";

import { FormRow } from './FormRow.js';
import { TextInput } from './TextInput.js';

export function TextInputRow({ name, label, text, error, ...props}) {
  return html`
<${FormRow} name=${name} label=${label} text=${text} error=${error} >
  <${TextInput} name="${name}" ...${props} />
</${FormRow}>`;
}
