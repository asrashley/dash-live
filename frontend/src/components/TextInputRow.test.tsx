import { afterEach, beforeEach, describe, expect, test, vi } from "vitest";

import { renderWithProviders } from "../test/renderWithProviders";
import { TextInputRow } from "./TextInputRow";
import { signal } from "@preact/signals";

describe("TextInputRow", () => {
  const name = "tirtest";
  const onInput = vi.fn();
  const error = signal<string|undefined>();

  beforeEach(() => {
    error.value = undefined;
  });

  afterEach(() => {
    vi.clearAllMocks();
  });

  test("should match snapshot", () => {
    const label = "the input label";
    const text = "description of the input";
    const { asFragment, getByText } = renderWithProviders(
      <TextInputRow
        name={name}
        onInput={onInput}
        placeholder="search..."
        required={true}
        label={label}
        text={text}
        error={error}
      />
    );
    getByText(label, { exact: false });
    getByText(text);
    expect(asFragment()).toMatchSnapshot();
  });

  test("shows error", () => {
    const label = "the input label";
    const text = "description of the input";
    error.value = "there is a problem with this value";
    const { getByText } = renderWithProviders(
      <TextInputRow
        name={name}
        onInput={onInput}
        placeholder="search..."
        required={true}
        label={label}
        text={text}
        error={error}
      />
    );
    getByText(label, { exact: false });
    getByText(text);
    getByText(error);
  });
});
