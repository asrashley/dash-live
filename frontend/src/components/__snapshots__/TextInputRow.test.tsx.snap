// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TextInputRow > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="row mb-2 form-group "
  >
    <label
      class="col-2 col-form-label"
      for="model-tirtest"
      id="label-tirtest"
    >
      the input label:
    </label>
    <div
      class="col-7"
    >
      <input
        class="form-control is-valid"
        id="field-tirtest"
        name="tirtest"
        placeholder="search..."
        required=""
        type="text"
      />
    </div>
    <div
      class="col-3 form-text"
      id="text-tirtest"
    >
      description of the input
    </div>
    <div
      class="invalid-feedback"
      style="display: none;"
    />
  </div>
</DocumentFragment>
`;
